typedb/connection/session.py

```diff
@@ -103,41 +96,33 @@ def close(self) -> None:
             self._rw_lock.acquire_write()
             if self._is_open.compare_and_set(True, False):
                 self._client.remove_session(self)
-                try:
-                    self._scheduler.cancel(self._pulse)
-                except ValueError:  # This may occur if a pulse is in transit right now.
-                    pass
                 req = session_proto.Session.Close.Req()
                 req.session_id = self._session_id
                 try:
                     self._stub().session_close(req)
-                except RpcError:  # This generally means the session is already closed.
+                except TypeDBClientException:  # This generally means the session is already closed.
```

Comment:
Fixes a bug where the client would throw if the session was already closed. It was just straight-up incorrect code - `RpcError` is never thrown by `session_close`.